h/static/scripts/guest.coffee

```diff
@@ -119,9 +120,27 @@ class Annotator.Guest extends Annotator
       # Announce the new positions, so that the sidebar knows
       this.plugins.Bridge.sync([highlight.annotation])
 
+  # Utility function to remove the hash part from a URL
+  _removeHash: (url) ->
+    url = new URL url
```

Comment:
> This API is not the most portable: https://developer.mozilla.org/en/docs/Web/API/URL#Browser_compatibility

True; but we are already using it - see [here](https://github.com/hypothesis/h/blob/a8811acdc63b014acc82d712d3dc83c4e8b9197b/h/static/scripts/annotator/plugin/pdf.coffee#L287)

> Can you conceive of an alternative or include a polyfill?

I'll look for a polyfill. (I would hate to use some partial workaround when there is a proper API for doing something.)